src/shared/varlink-serialize.c

```diff
@@ -84,3 +85,17 @@ int varlink_server_deserialize_one(sd_varlink_server *s, const char *value, FDSe
         LIST_PREPEND(sockets, s->sockets, TAKE_PTR(ss));
         return 0;
 }
+
+int varlink_server_contains_socket(sd_varlink_server *s, const char *address) {
+        assert(s);
+        assert(address);
+
+        LIST_FOREACH(sockets, ss, s->sockets)
+                if (address[0] == '@') {
+                        if (streq_ptr(ss->address, address))
+                                return true;
+                } else if (path_equal(ss->address, address))
+                        return true;
```

Comment:
hmm, i'd like this to be a generic thing, maybe in socket-util.c, close to socket_address_parse_unix(), because it should follow the same rules.

i.e. a helper function socket_address_equal_unix() or so?

Reply:
(could even use socket_address_parse_unix() on both args, and then use sockaddr_equal())